### PR TITLE
Various: Scattered trivial cleanups

### PR DESCRIPTION
**Pull request by antonlacon**
These are some cleanup commits that have been floating around my branches. All of them are fairly trivial, so I grouped them up.

rpi: housekeeping edits comments to reflect the options available to the RPi.

projects: ratpoison removes references to the ratpoison wm from the project comments. I believe ratpoison was dropped a couple years ago from the tree.

config/options: replaces "which" with "command -v". Command -v is POSIX and has the same result as which. nproc is part of coreutils, and returns the number of processing units available, which is what the grep expression was up to.

scripts/checkdeps: which to command -v, and be more graceful when the 'sudo' command isn't present. I don't have sudo on my machines.

### CODE REVIEW

File: scripts/checkdeps [hunk 1]
```diff
@@ -132,34 +132,38 @@ if [ "${#need[@]}" -gt 0 ]; then
...
-      exit 1
-      ;;
-  esac
+  if command -v sudo >/dev/null; then
+    case "$DISTRO" in
+      ubuntu|debian|linuxmint|\"elementary\")
+        get_yes_no && sudo apt-get install "${need_pkg[@]}"
+        ;;
+      fedora|centos|rhel)
+        if command -v dnf >/dev/null; then YUM=dnf; else YUM=yum; fi
```
Comment (MilhouseVH) [inline scripts/checkdeps]:
```
command -v dnf >/dev/null && YUM=dnf || YUM=yum
```
?

**Comment by MilhouseVH** [comment]
Other than the small nit this looks good to me, thanks.

**Comment by antonlacon** [comment]
config/functions: pair of what I believe to be unused functions anywhere. eglibc was rolled into glibc proper a few years ago. tolower() has been around for years and just appears unused. It could be improved if it stays.

File: scripts/checkdeps (hunk 1 above)
Comment (antonlacon) [reply scripts/checkdeps]: Looks good to me.

**Comment by antonlacon** [comment]
mkpkg_lockdev: drop as lockdev was dropped from tree early August.

**Pull request merged**